### PR TITLE
refactor(Collectibles): collectible list

### PR DESCRIPTION
**Pull request by UncleBill**
- auto even columns
- align loading skeleton and collectible item

### CODE REVIEW

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        position: 'relative',
+        padding: theme.spacing(1, 0),
+    },
+    collectibleCard: {
+        width: '100%',
+        height: '100%',
+        aspectRatio: '1/1',
+    },
+    description: {
+        background: theme.palette.mode === 'light' ? '#F7F9FA' : '#2F3336',
```
Comment (guanbinrui) [inline packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx]: Recommend using a theme color.

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleCard.tsx [hunk 2]
```diff
@@ -15,8 +15,8 @@ const useStyles = makeStyles()((theme) => ({
         position: 'absolute',
         zIndex: 1,
         backgroundColor: theme.palette.mode === 'light' ? '#F7F9FA' : '#2F3336',
```
Comment (guanbinrui) [inline packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleCard.tsx:17]: Recommend using a theme color.

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx (hunk 1 above)
Comment (guanbinrui) [inline packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx]: Recommend using a theme color.

Comment (UncleBill) [reply packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx]: Done.

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        position: 'relative',
+        padding: theme.spacing(1, 0),
+    },
+    collectibleCard: {
+        width: '100%',
+        height: '100%',
+        aspectRatio: '1/1',
+    },
+    description: {
+        background: theme.palette.mode === 'light' ? MaskColorVar.twitterBackground : MaskColorVar.twitterLine,
```
Comment (guanbinrui) [inline packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx]: I think the `MaskColorVar` is deprecated. CC @nuanyang233

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    },
+}))
+
+interface CollectibleItemProps extends HTMLProps<HTMLDivElement> {
+    provider: SourceType
+    wallet?: Wallet
+    asset: NonFungibleAsset<Web3Helper.ChainIdAll, Web3Helper.SchemaTypeAll>
+    readonly?: boolean
+    renderOrder: number
+    address?: SocialAddress<NetworkPluginID>
```
Comment (guanbinrui) [inline packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx]: We can export the `CollectibleCardProps` here just as the MUI usually did. It gives more refined control to parent components and reduces coding labor.

```ts
export interface CollectibleItemProps {
    CollectibleCardProps?: Partial<CollectibleCardProps>
}

export function CollectibleItem(props: CollectibleItemProps) {
    return (
        <CollectibleCard
            {...props.CollectibleCardProps}
        />
    )
}
```

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx (hunk 3 above)
Comment (UncleBill) [reply packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx]: It's deprecated, but it's also hard to tell what color can match. There are still many usages in the codebase.

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    CollectibleCardProps?: Partial<CollectibleCardProps>
+}
+export function CollectibleItem(props: CollectibleItemProps) {
+    const { provider, wallet, asset, readonly, renderOrder, address, className, ...rest } = props
```
Comment (guanbinrui) [inline packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx:37]:
```suggestion
    const { className, ...rest } = props
```

Please move their props to `CollectibleCardProps`.

File: packages/plugins/RSS3/src/SNSAdaptor/pages/DonationsPage.tsx [hunk 6]
```diff
@@ -60,19 +60,20 @@ export function DonationPage({ socialAddress, publicKey, userId }: DonationPageP
...
     const donations = useAvailableCollections(
         kvValue?.proofs ?? EMPTY_LIST,
-        CollectionType.Donations,
         allDonations,
+        CollectionType.Donations,
         userId,
         socialAddress.address,
     )
 
+    const twitterId = userId.toLowerCase()
```
Comment (guanbinrui) [inline packages/plugins/RSS3/src/SNSAdaptor/pages/DonationsPage.tsx]: This plugin should also work for other SNS websites. Not a proper variable name.

File: packages/plugins/RSS3/src/SNSAdaptor/index.tsx [hunk 7]
```diff
@@ -16,7 +16,7 @@ const DonationsTabConfig: Plugin.SNSAdaptor.ProfileTab = {
     priority: 1,
     UI: {
         TabContent: ({ socialAddress, identity }) => {
-            if (!socialAddress?.address || !identity?.identifier?.userId || !identity.publicKey) return null
+            if (!socialAddress?.address || !identity?.identifier?.userId) return null
```
Comment (guanbinrui) [inline packages/plugins/RSS3/src/SNSAdaptor/index.tsx]: Does this condition mean a social identity must be a NextID user? Please consult QA or PM.

File: packages/plugins/RSS3/src/SNSAdaptor/hooks/useAvailableCollections.ts [hunk 8]
```diff
@@ -6,20 +6,20 @@ import type { Proof } from '../../types'
 
 export const useAvailableCollections = (
     proofs: Proof[],
-    type: CollectionType,
     collections: RSS3BaseAPI.Collection[],
-    userId?: string,
+    type: CollectionType,
+    twitterId?: string,
```
Comment (guanbinrui) [inline packages/plugins/RSS3/src/SNSAdaptor/hooks/useAvailableCollections.ts]: The `userId` is more accurate than this one.

File: packages/mask/src/plugins/Collectible/SNSAdaptor/NFTPage.tsx [hunk 9]
```diff
@@ -1,22 +1,19 @@
...
-    const currentVisitingProfile = useCurrentVisitingProfile()
-
     if (!socialAddress) return null
 
     return (
         <CollectionList
             addressName={socialAddress}
             persona={identity?.publicKey}
-            visitingProfile={currentVisitingProfile}
+            twitterId={identity?.identifier?.userId.toLowerCase()}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/NFTPage.tsx]: I think the original edition is better. All NextID applications should start from a `SocialIdentity`.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**